### PR TITLE
Type property inheritance

### PR DESCRIPTION
**Pull request by AshleySetter**
Implemented type inheritance

- Already defined properties overload inherited properties
- Properties inherited must be specified
- Properties are passed down the tree from root nodes to leaf nodes (node A can inherit a property from B, which it inherited from C and so on)
- Throws an error if:
  - parent type doesn't exist
  - duplicate properties are defined as inherited from parents
  - property specified to inherit from parent is missing from parent

To Do:
- Need to implement tests

### CODE REVIEW

**Review by cbrafter — APPROVED** [review]
Looks great!

File: d3-scripts/tests/__fixtures__/duplicate-property-type-inheritance-single-parent/test-2.type.d3.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+credentialSubject:
+  # unique identifiers
+  id: e22b0268-b6ca-42f8-9317-b629e12286ef
+
+  parents:
+    - id: ba3ba36d-be70-40f0-9c52-95b560066d11 # Id of parent type to inherit from
+      # Specific properties to inherit
+      properties:
+        - manufacturer
+        - manufacturer
```
Comment (cbrafter) [inline d3-scripts/tests/__fixtures__/duplicate-property-type-inheritance-single-parent/test-2.type.d3.yaml:11]: Duplicate?

**Review by cbrafter — APPROVED** [review]
Great work!

File: d3-scripts/tests/__fixtures__/duplicate-property-type-inheritance-single-parent/test-2.type.d3.yaml (hunk 1 above)
Comment (AshleySetter) [reply d3-scripts/tests/__fixtures__/duplicate-property-type-inheritance-single-parent/test-2.type.d3.yaml:11]: yeah, it's on purpose, this test claim is checking an error is thrown when duplicate properties are entered :smile:

**Pull request merged**